microprofile/WFLY-12685_upgrade_microprofile_health_2.1.0.adoc

```diff
@@ -0,0 +1,62 @@
+= Upgrade MicroProfile Health to 2.1.0
+:author:            Jeff Mesnil
+:email:             jmesnil@redhat.com
+:toc:               left
+:icons:             font
+:keywords:          microprofile,health,observability
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+
+MicroProfile 3.1 provides a minor upgrade for https://github.com/eclipse/microprofile-health/releases/tag/2.1[MicroProfile Health 2.1.0].
+
+This component upgrade also covers the upgrade of smallrye-health 2.2.0 that implements MicroProfile Health 2.1.0.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-12685[WFLY-12685]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/WFLY-12734[WFLY-12734] - Upgrade smallrye-health to 2.2.0
+* https://issues.jboss.org/browse/EAP7-1348[EAP7-1348]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:istraka@redhat.com[Ivan Straka]
+
+=== Testing By
+
+[X] QE
+
+== Requirements
+
+* Upgrade MicroProfile Health to 2.1.0
+* Upgrade smallrye-health to 2.2.0
+* Run any MicroProfile Health test in WildFly integration test suite
+* Run the Eclipse MicroProfile Health TCK without failures
+
+=== Non-Requirements
+
+* Support of the `mp.health.disable-default-procedures` config property is postponed to https://issues.jboss.org/browse/WFLY-12342[WFLY-12342] that will add default server procedures.
+* Porting of https://github.com/thorntail-examples/health-check/blob/f164faa787d4adb616424116a1ec84a1f28b30eb/src/main/java/io/thorntail/example/HealthChecks.java#L31[test for Thorntail health-check example] is out of scope and will be done in the context of https://issues.jboss.org/browse/WFLY-12342[WFLY-12342] that will adds a proper server probes to check the `server-state` status.
+
+== Implementation Plan
+
+== Test Plan
+
+* Run the WildFly integration basic test suite and checks there are no failures related to MicroProfle Health tests.
+* Run the `microprofile-tck/health` module from WildFly integration test suite and checks that there are no failures.
+* There are no existing tests for MicroProfile Health in https://github.com/thorntail/thorntail/tree/master/testsuite[Thorntail test suite] that needs to be ported to WildFly integration test suite.
```

Comment:
There is [a ParserTest.java](https://github.com/thorntail/thorntail/blob/master/fractions/microprofile/microprofile-health/src/test/java/org/wildfly/swarm/health/ParserTest.java) as Rosta mentioned in [a comment](https://issues.jboss.org/browse/EAP7-1348?focusedCommentId=13807656&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13807656). There are two simple tests.

Reply:
@istraka thanks for the link.

WildFly test suite and MP Health TCK already covers such basic tests in https://github.com/wildfly/wildfly/tree/master/testsuite/integration/basic/src/test/java/org/wildfly/test/integration/microprofile/health.
I don't see any benefit to add this test.

Reply:
I've updated the dev analysis to mention that this test will not be added to WildFly test suite